Java/core/src/main/java/com/amazon/randomcutforest/state/RandomCutForestState.java

```diff
@@ -48,8 +48,22 @@
 
     private boolean compactEnabled;
 
+    private boolean internalShinglingEnabled;
+
+    private boolean internalRotationEnabled;
+
+    private boolean dynamicResizingEnabled;
+
     private boolean centerOfMassEnabled;
 
+    private boolean saveTreeState;
+
+    private boolean saveSamplerState;
+
+    private boolean saveCoordinatorState;
+
+    private boolean singlePrecisionSet;
```

Comment:
Why not use the Precision enum here? I don't think any serialization library will have trouble encoding an enum value.

Reply:
There are only two choices of the precision, we can go either way. Enums can help accommodate future point types -- but we should make is a point type enum?